src/dataExplorer/components/BucketSelector.tsx

```diff
@@ -1,51 +1,174 @@
-import React, {FC, useContext, useState} from 'react'
+import React, {FC, Fragment, ChangeEvent, useContext, useState} from 'react'
+import {
+  TechnoSpinner,
+  ComponentSize,
+  Dropdown,
+  DropdownMenuTheme,
+  IconFont,
+  Input,
+} from '@influxdata/clockface'
 
 // Components
-import {ComponentStatus} from '@influxdata/clockface'
 import SelectorTitle from 'src/dataExplorer/components/SelectorTitle'
-import SearchableDropdown from 'src/shared/components/SearchableDropdown'
 
 // Contexts
 import {FluxQueryBuilderContext} from 'src/dataExplorer/context/fluxQueryBuilder'
 import {BucketContext} from 'src/shared/contexts/buckets'
 import {event} from 'src/cloud/utils/reporting'
 
+// Types
+import {RemoteDataState, Bucket} from 'src/types'
+
 const BUCKET_TOOLTIP = `A bucket is a named location where time series data \
 is stored. You can think of a bucket like you would a database in SQL systems.`
 
+const REMAP_BUCKET_TYPES = {
+  user: 'My Data',
+  system: 'System Data',
+  sample: 'Sample Data',
+}
+
 const BucketSelector: FC = () => {
   const {selectedBucket, selectBucket} = useContext(FluxQueryBuilderContext)
-  const {buckets} = useContext(BucketContext)
+  const {loading, buckets} = useContext(BucketContext)
+  const [isSearchActive, setIsSearchActive] = useState(false)
   const [searchTerm, setSearchTerm] = useState('')
 
-  const handleSelectBucket = (name: string) => {
-    const bucket = buckets.find(b => b.name === name)
-    if (!bucket) {
-      return
-    }
+  const _buckets = buckets.filter(b =>
+    `${b.name}`.toLocaleLowerCase().includes(searchTerm.toLocaleLowerCase())
+  )
+
+  const handleSelectBucket = (buck: Bucket) => {
     event('bucketSelected', {search: searchTerm.length})
-    selectBucket(bucket)
+    selectBucket(buck)
+  }
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>): void => {
+    setSearchTerm(e.target.value)
+  }
+
+  let buttonText = 'Loading buckets...'
+  if (loading === RemoteDataState.Done && !selectedBucket?.name) {
+    buttonText = 'Choose a bucket'
```

Comment:
do we want to keep the button text consistent with the [design](https://www.figma.com/proto/0qAntPk5LVangAHguWT74X/Query-Experience-Project?page-id=191%3A13710&node-id=567%3A21115&viewport=241%2C48%2C0.15&scaling=min-zoom&starting-point-node-id=567%3A21115) using "Select bucket..."?